web/pf4/src/utils/victoryChartsUtils.ts

```diff
@@ -58,3 +58,27 @@ export const getDataSupplier = (chart: ChartModel, labelValues: AllPromLabelsVal
   }
   return undefined;
 };
+
+export const buildLegend = (rawLegend: string[], chartWidth: number): LegendInfo => {
+  // Very arbitrary rules to try to get a good-looking legend. There's room for enhancement.
+  const items = rawLegend.map(it => ({ name: it }));
+  // Box size in pixels per item
+  // Note that it is based on longest string in characters, not pixels
+  let boxSize = 110;
+  const longest = items.map(it => it.name).reduce((a, b) => a.length > b.length ? a : b, '').length;
+  if (longest >= 30) {
+    boxSize = 400;
+  } else if (longest >= 20) {
+    boxSize = 300;
+  } else if (longest >= 10) {
+    boxSize = 200;
+  }
+  const itemsPerRow = Math.max(1, Math.floor(chartWidth / boxSize));
+  const nbRows = Math.floor((items.length - 1) / itemsPerRow) + 1;
```

Comment:
Hmm... did you check if

```js
const nbRows = Math.ceil(items.length / itemsPerRow);
```

would give the same result? (just because it looks simpler)